### ISSUE TITLE
Fix rubocop issues

### ISSUE THREAD

**Issue by athityakumar** [SciRuby/networkx.rb#17]
### Description

Following PR #16, there are a couple of rubocop linter issues that need to be fixed 

- Relevant Issues : (optional)
- Relevant PRs : #16 
- Type of issue :
  - User mode :
    - [ ] Installation
    - [ ] Feature request
    - [ ] Bug in existing feature
  - Developer mode :
    - [x] Code quality
    - [ ] Tests
    - [ ] Documentation

**Comment by arzoo14** [comment]
@athityakumar
I want to fix this issue. Can you please assign it to me ? Please also guide for those rubocop offenses in which module length , complexity increases as in those I have to change some piece of code as well !

**Comment by athityakumar** [comment]
Sure @arzoo14, go ahead with fixing this issue! 👍

Most of the complexity-related issues can be fixed by splitting existing functions into smaller and meaningful modular functions. To start off, you can try running `bundle exec rubocop` after un-commenting all `Rubocop:disable` comments in the codebase and check which functions face these issues.

If you come across an issue with a specific function, please open a PR and we can discuss there. 😄

**Comment by athityakumar** [comment]
PR #19 has partially taken care of some rubocop issues, but the main complexity issues are still existing. So, I'm keeping this issue open till it gets solved too.

**Comment by universato** [comment]
Done!
I suppressed all rubocop's alerts by editing some code and disabling some cops. 

- Metrics Cops warnings are hard to resolve.
- Splitting into multiple methods doesn't always improve readability.
- Short methods don't always improve readability.
- RuboCop is not always correct.
- Too many warnings delay finding real bugs.

For these reasons, I disabled Metrics Cop.
In actual some of the remain rubocops' warnings indicated bugs, so I fixed these bugs.
